doc: Properly report optional RPC args

maflcko opened the pull request:
`OMITTED_NAMED_ARG` and `OMITTED` are a confusing burden:

* It puts the burden on developers to pick the right one of the two
* They can be interchanged without introducing a compile failure or other error
* Picking the wrong one is leading to incorrect docs
* They are redundant, because the correct one can already be determined by the surrounding type

Fix all issues by making them an alias of each other; Pick the right one based on the outer type.

MarcoFalke commented:
<details><summary>Rendered RPC doc diff:</summary>

```diff
diff --git a/importdescriptors b/importdescriptors
index e8d2dd2..44e9ef0 100644
--- a/importdescriptors
+++ b/importdescriptors
@@ -12,8 +12,8 @@ Arguments:
        {                                    (json object)
          "desc": "str",                     (string, required) Descriptor to import.
          "active": bool,                    (boolean, optional, default=false) Set this descriptor to be the active descriptor for the corresponding output type/externality
-         "range": n or [n,n],               (numeric or array) If a ranged descriptor is used, this specifies the end or the range (in the form [begin,end]) to import
-         "next_index": n,                   (numeric) If a ranged descriptor is set to active, this specifies the next index to generate addresses from
+         "range": n or [n,n],               (numeric or array, optional) If a ranged descriptor is used, this specifies the end or the range (in the form [begin,end]) to import
+         "next_index": n,                   (numeric, optional) If a ranged descriptor is set to active, this specifies the next index to generate addresses from
          "timestamp": timestamp | "now",    (integer / string, required) Time from which to start rescanning the blockchain for this descriptor, in UNIX epoch time
                                             Use the string "now" to substitute the current synced blockchain time.
                                             "now" can be specified to bypass scanning, for outputs which are known to never have been used, and
diff --git a/importmulti b/importmulti
index 6433ec3..aa238e2 100644
--- a/importmulti
+++ b/importmulti
@@ -14,7 +14,7 @@ Arguments:
 1. requests                                                         (json array, required) Data to be imported
      [
        {                                                            (json object)
-         "desc": "str",                                             (string) Descriptor to import. If using descriptor, do not also provide address/scriptPubKey, scripts, or pubkeys
+         "desc": "str",                                             (string, optional) Descriptor to import. If using descriptor, do not also provide address/scriptPubKey, scripts, or pubkeys
          "scriptPubKey": "<script>" | { "address":"<address>" },    (string / json, required) Type of scriptPubKey (string for script, json for address). Should not be provided if using a descriptor
          "timestamp": timestamp | "now",                            (integer / string, required) Creation time of the key expressed in UNIX epoch time,
                                                                     or the string "now" to substitute the current synced blockchain time. The timestamp of the oldest
@@ -22,8 +22,8 @@ Arguments:
                                                                     "now" can be specified to bypass scanning, for keys which are known to never have been used, and
                                                                     0 can be specified to scan the entire blockchain. Blocks up to 2 hours before the earliest key
                                                                     creation time of all keys being imported by the importmulti call will be scanned.
-         "redeemscript": "str",                                     (string) Allowed only if the scriptPubKey is a P2SH or P2SH-P2WSH address/scriptPubKey
-         "witnessscript": "str",                                    (string) Allowed only if the scriptPubKey is a P2SH-P2WSH or P2WSH address/scriptPubKey
+         "redeemscript": "str",                                     (string, optional) Allowed only if the scriptPubKey is a P2SH or P2SH-P2WSH address/scriptPubKey
+         "witnessscript": "str",                                    (string, optional) Allowed only if the scriptPubKey is a P2SH-P2WSH or P2WSH address/scriptPubKey
          "pubkeys": [                                               (json array, optional, default=[]) Array of strings giving pubkeys to import. They must occur in P2PKH or P2WPKH scripts. They are not required when the private key is also provided (see the "keys" argument).
            "pubKey",                                                (string)
            ...
@@ -32,7 +32,7 @@ Arguments:
            "key",                                                   (string)
            ...
          ],
-         "range": n or [n,n],                                       (numeric or array) If a ranged descriptor is used, this specifies the end or the range (in the form [begin,end]) to import
+         "range": n or [n,n],                                       (numeric or array, optional) If a ranged descriptor is used, this specifies the end or the range (in the form [begin,end]) to import
          "internal": bool,                                          (boolean, optional, default=false) Stating whether matching outputs should be treated as not incoming payments (also known as change)
          "watchonly": bool,                                         (boolean, optional, default=false) Stating whether matching outputs should be considered watchonly.
          "label": "str",                                            (string, optional, default="") Label to assign to the address, only allowed with internal=false
diff --git a/scanblocks b/scanblocks
index 017b89b..e88e00c 100644
--- a/scanblocks
+++ b/scanblocks
@@ -8,7 +8,7 @@ Arguments:
                                  "start" for starting a scan
                                  "abort" for aborting the current scan (returns true when abort was successful)
                                  "status" for progress report (in %) of the current scan
-2. scanobjects                   (json array) Array of scan objects. Required for "start" action
+2. scanobjects                   (json array, optional) Array of scan objects. Required for "start" action
                                  Every scan object is either a string descriptor or an object:
      [
        "descriptor",             (string) An output descriptor
diff --git a/scantxoutset b/scantxoutset
index 31ed769..654c806 100644
--- a/scantxoutset
+++ b/scantxoutset
@@ -19,7 +19,7 @@ Arguments:
                                  "start" for starting a scan
                                  "abort" for aborting the current scan (returns true when abort was successful)
                                  "status" for progress report (in %) of the current scan
-2. scanobjects                   (json array) Array of scan objects. Required for "start" action
+2. scanobjects                   (json array, optional) Array of scan objects. Required for "start" action
                                  Every scan object is either a string descriptor or an object:
      [
        "descriptor",             (string) An output descriptor
diff --git a/signrawtransactionwithkey b/signrawtransactionwithkey
index cd163d9..d042181 100644
--- a/signrawtransactionwithkey
+++ b/signrawtransactionwithkey
@@ -19,9 +19,9 @@ Arguments:
          "txid": "hex",             (string, required) The transaction id
          "vout": n,                 (numeric, required) The output number
          "scriptPubKey": "hex",     (string, required) script key
-         "redeemScript": "hex",     (string) (required for P2SH) redeem script
-         "witnessScript": "hex",    (string) (required for P2WSH or P2SH-P2WSH) witness script
-         "amount": amount,          (numeric or string) (required for Segwit inputs) the amount spent
+         "redeemScript": "hex",     (string, optional) (required for P2SH) redeem script
+         "witnessScript": "hex",    (string, optional) (required for P2WSH or P2SH-P2WSH) witness script
+         "amount": amount,          (numeric or string, optional) (required for Segwit inputs) the amount spent
        },
        ...
      ]
diff --git a/signrawtransactionwithwallet b/signrawtransactionwithwallet
index e41e3d0..22d1a2c 100644
--- a/signrawtransactionwithwallet
+++ b/signrawtransactionwithwallet
@@ -13,9 +13,9 @@ Arguments:
          "txid": "hex",             (string, required) The transaction id
          "vout": n,                 (numeric, required) The output number
          "scriptPubKey": "hex",     (string, required) script key
-         "redeemScript": "hex",     (string) (required for P2SH) redeem script
-         "witnessScript": "hex",    (string) (required for P2WSH or P2SH-P2WSH) witness script
-         "amount": amount,          (numeric or string) (required for Segwit inputs) the amount spent
+         "redeemScript": "hex",     (string, optional) (required for P2SH) redeem script
+         "witnessScript": "hex",    (string, optional) (required for P2WSH or P2SH-P2WSH) witness script
+         "amount": amount,          (numeric or string, optional) (required for Segwit inputs) the amount spent
        },
        ...
      ]
```

</details>

Commit:
refactor: Remove const to fix performance-move-const-arg clang-tidy errors

The warnings look like:

src/rpc/util.h:192:19: error: std::move of the const variable 'name' has no effect; remove std::move() or make the variable non-const [performance-move-const-arg,-warnings-as-errors]
        : m_names{std::move(name)},
                  ^~~~~~~~~~    ~

Commit:
doc: Properly report optional RPC args

Commit:
refactor: Introduce is_top_level_arg

hebasto commented:
FWIW, the first commit has been cherry-picked into #26707.

fanquake reviewed (approved):
ACK fad56f7dd6ecac772e1bf590b38cd34ba67db5d7

kouloumos commented on src/rpc/util.h:
```diff
@@ -154,21 +154,24 @@ struct RPCArg {
+         * `null`.
+         *
          * Optional argument with default value omitted because they are
-         * implicitly clear. That is, elements in an array or object may not
```
Is there a reason that "or object" was removed from here? My understanding is that `RPCArg::Optional::OMITTED` still applies to elements in objects, such as here
https://github.com/bitcoin/bitcoin/blob/392dc68e37be9fc7adb32496b13d9b50262e317d/src/wallet/rpc/backup.cpp#L1258-L1293

MarcoFalke replied:
"or object" is covered by the first of the two reasons:

> Optional arg that is a named argument and has a default value of null


(named arguments are basically keys in a dictionary/object)

kouloumos replied:
Thank you, the root of my confusion was the definition of named arguments.

MarcoFalke replied:
If there are improvements, they can be appended to #26919, which is a follow-up
